File: src/java/burp/model/Return.java
package burp.model;

import java.util.HashMap;
import java.util.Map;

public class Return {

	private Map<String, Object> returns = new HashMap<String, Object>();

	public Object defaultValue = null;
	
	public Return(Object defaultValue) {
		this.defaultValue = defaultValue;
	}

	public Object get(Object key) {
		Object o = returns.get(key);
		
		if(o == null)
			throw new RuntimeException("Unknown return value %s.".formatted(key));
		
		return o;
	}

	public Object put(String key, Object value) {
		return returns.put(key, value);
	}
	
}

File: src/main/java/burp/model/fnmlutil/Return.kt
package burp.model.fnmlutil


class Return(defaultValue: Any?) {
    private val returns = mutableMapOf<String, Any?>()

    @JvmField
    var defaultValue: Any? = null

    init {
        this.defaultValue = defaultValue
    }

    fun get(key: String) = returns.getOrElse(key) {
        throw RuntimeException("ExecutionError: Unknown return value $key.")
    }


    fun put(key: String, value: Any?) = returns.put(key, value)

}
